### PR TITLE
Large data - follow-up/phase 2 work

### PR DESCRIPTION
**Pull request by ukengineer925**
This PR contains follow-on work identified as non-critical fixes or wider changes from #204.

### CODE REVIEW

**Review by drivenflywheel — CHANGES_REQUESTED** [review]
This PR needs to have **master** specified as its target branch; it currently has **large-data-phase-1** specified as its target.

**Comment by drivenflywheel** [comment]
See PR #363 as a replacement for this PR

File: src/main/java/emissary/core/channels/InputStreamChannelFactory.java [hunk 1]
```diff
@@ -69,7 +69,7 @@ public InputStreamChannel(final long size, final InputStreamFactory inputStreamF
         }
 
         @Override
-        protected final int readImpl(final ByteBuffer byteBuffer) throws IOException {
+        protected final int readImpl(final ByteBuffer byteBuffer, final int maxBytesToRead) throws IOException {
             if (position() < currentPosition) {
```
[the remark this answers is not in the document]
Comment (ukengineer925) [reply src/main/java/emissary/core/channels/InputStreamChannelFactory.java:73]: Can certainly rename currentPosition -> streamPosition. The lack of validation for size is intentional, if a negative value is provided the size is determined upon the first `size()` call by reading through the inputStream to count the bytes.

File: src/test/java/emissary/core/channels/AbstractSeekableByteChannelTest.java [hunk 2]
```diff
@@ -40,65 +45,76 @@ void testConstrutorValidSize() throws IOException {
...
         try (SeekableByteChannel sbc = new TestSeekableByteChannel(10)) {
-            Assertions.assertThrows(NullPointerException.class, () -> sbc.read(null));
+            assertThrows(NullPointerException.class, () -> sbc.read(null));
 
             final ByteBuffer byteBuffer = ByteBuffer.allocate(4);
 
-            Assertions.assertEquals(0, sbc.position());
-            Assertions.assertEquals(4, sbc.read(byteBuffer));
+            assertEquals(0, sbc.position());
+            assertEquals(4, sbc.read(byteBuffer));
```
Comment (drivenflywheel) [inline src/test/java/emissary/core/channels/AbstractSeekableByteChannelTest.java:102]: Should also include a test for which there are fewer "content" bytes available in the channel than the size of the byte buffer.  For instance, for this specific channel, set the position to `sbc.size() - 2`, and assert that `sbc.read(byteBuffer)` returns 2.  That helps validate the behavior of the `maxBytesToRead` implementation.

**Review by drivenflywheel — CHANGES_REQUESTED** [review]
I like the changes in this PR.  One minor bug to address, but my other comments are more personal preference than problems.

File: src/main/java/emissary/core/IBaseDataObjectHelper.java [hunk 3]
```diff
@@ -113,15 +117,15 @@ public static IBaseDataObject clone(final IBaseDataObject iBaseDataObject, final
      *         underlying field is either inaccessible or final.
      * @throws NoSuchFieldException if a field with the specified name is not found.
      */
-    public static void setPrivateField(final BaseDataObject bdo, final String fieldName, final Object object)
+    public static void setPrivateFieldValue(final BaseDataObject bdo, final String fieldName, final Object object)
```
[the remark this answers is not in the document]
Comment (ukengineer925) [reply src/main/java/emissary/core/IBaseDataObjectHelper.java:120]: I've created a ticket to log addressing this in the future.

File: src/test/java/emissary/core/channels/InputStreamChannelFactoryTest.java [hunk 4]
```diff
@@ -97,11 +120,34 @@ void testReadWithIncorrectSize() throws IOException {
...
         }
     }
+
+    @Test
+    void testStartReadBeyondActualData() throws IOException {
+        // Try reading beyond the actual data
+        try (final SeekableByteChannel sbc = InputStreamChannelFactory.create(20, new TestInputStreamFactory(testBytes)).create()) {
+            ByteBuffer buff = ByteBuffer.allocate(32);
+            assertEquals(20, sbc.size());
+            sbc.position(12);
```
[the remark this answers is not in the document]
Comment (ukengineer925) [reply src/test/java/emissary/core/channels/InputStreamChannelFactoryTest.java]: Addressed in new commit

File: src/test/java/emissary/core/channels/AbstractSeekableByteChannelTest.java (hunk 2 above)
Comment (ukengineer925) [reply src/test/java/emissary/core/channels/AbstractSeekableByteChannelTest.java:100]: Also now addresses in latest commit.

**Review by drivenflywheel — APPROVED** [review]
LGTM

**Review by jpdahlke — APPROVED** [review]

File: src/main/java/emissary/core/IBaseDataObjectHelper.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (jpdahlke) [reply src/main/java/emissary/core/IBaseDataObjectHelper.java:120]: https://github.com/NationalSecurityAgency/emissary/pull/656

**Pull request merged**